src/main/java/core/basesyntax/utils/generate/ReportGenerator.java

```diff
@@ -0,0 +1,7 @@
+package core.basesyntax.utils.generate;
+
+import core.basesyntax.storage.impl.StorageImpl;
+
+public interface ReportGenerator {
+    String getReport(StorageImpl storage);
```

Comment:
According to the SOLID principles, specifically the Dependency Inversion Principle, we should depend upon abstractions, not concretions. It's better to pass a more generic type like `Storage` rather than the concrete implementation `StorageImpl`.